File: sdks/node/examples/offline-verify.js
import { verifyTrustTokenOffline, fetchRevocations } from '../src/trust.js'

async function main() {
  const baseURL = process.env.AURA_API_BASE_URL || 'http://localhost:8081';
  const orgId = process.env.AURA_ORG_ID || 'org_00000000000000000000000000000000';
  const token = process.env.AURA_TRUST_TOKEN;
  if (!token) {
    console.error('Set AURA_TRUST_TOKEN to a JWT to verify');
    process.exit(1);
  }
  let revoked = new Set();
  try {
    const { items } = await fetchRevocations({ baseURL, orgId });
    revoked = new Set(items.map(i => i.jti));
  } catch {}
  const res = await verifyTrustTokenOffline({ baseURL, token, orgId, graceSeconds: 30, revokedJTIs: revoked });
  console.log(res);
}

main().catch((e) => { console.error(e); process.exit(1); })

File: sdks/python/examples/offline_verify.py
import os
import time
from aura_sdk import verify_trust_token_offline, TrustCaches

base = os.environ.get("AURA_API_BASE_URL", "http://localhost:8081")
org = os.environ.get("AURA_ORG_ID")
token = os.environ.get("AURA_TRUST_TOKEN")

if not token:
    print("Set AURA_TRUST_TOKEN to a token to verify")
    raise SystemExit(2)

cache = TrustCaches(jwks_ttl=300, rev_ttl=60)

res = verify_trust_token_offline(token, base_url=base, org_id=org, grace_seconds=10, cache=cache)
if not res.get("valid"):
    print("invalid:", res.get("reason"))
    raise SystemExit(1)
print("valid claims:", res.get("claims"))
